Content.Server/DeltaV/Paper/SignatureSystem.cs

```diff
@@ -0,0 +1,103 @@
+using Content.Server.Access.Systems;
+using Content.Server.Paper;
+using Content.Server.Popups;
+using Content.Shared.Paper;
+using Content.Shared.Popups;
+using Content.Shared.Tag;
+using Content.Shared.Verbs;
+using Robust.Server.Audio;
+using Robust.Shared.Player;
+
+namespace Content.Server.DeltaV.Paper;
+
+public sealed class SignatureSystem : EntitySystem
+{
+    [Dependency] private readonly TagSystem _tagSystem = default!;
+    [Dependency] private readonly PaperSystem _paper = default!;
+    [Dependency] private readonly IdCardSystem _idCard = default!;
+    [Dependency] private readonly PopupSystem _popup = default!;
+    [Dependency] private readonly AudioSystem _audio = default!;
+
+    // The sprite used to visualize "signatures" on paper entities.
+    private static readonly string SignatureStampState = "paper_stamp-signature";
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<PaperComponent, GetVerbsEvent<AlternativeVerb>>(OnGetAltVerbs);
+    }
+
+    private void OnGetAltVerbs(EntityUid uid, PaperComponent component, GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract)
+            return;
+
+        var pen = args.Using;
+        if (pen == null || !_tagSystem.HasTag(pen.Value, "Write"))
+            return;
+
+        AlternativeVerb verb = new()
+        {
+            Act = () =>
+            {
+                TrySignPaper(uid, args.User, component);
+            },
+            Text = Loc.GetString("paper-sign-verb"),
+            DoContactInteraction = true,
+            Priority = 10
+        };
+        args.Verbs.Add(verb);
+    }
```

Comment:
the verb could easily be predicted by moving this to shared

Reply:
How do I do that? Paper components are different on the server and the client, and supposedly events do not support polymorphism, so I cannot subscribe on SharedPaperComponent.

Reply:
well if paper component isnt properly in shared upstream theres nothing that can be done realistically, dont worry about it